docs/source/composer_model.rst

```diff
@@ -0,0 +1,248 @@
+|:pickup_truck:| ComposerModel
+==============================
+
+Your Pytorch model and training step must be re-organized as a
+:class:`.ComposerModel` in order to use our :class:`.Trainer`.
+This interface helps our trainer access the necessary parts of your model
+to easily speed up training.
+
+Minimal Example
+---------------
+
+To create a trainable torchvision ResNet-18 classifier with a cross-entropy loss,
+define the |forward| and |loss| methods.
+
+Notice how the forward pass is still under user control (no magic here!)
+and encapsulated together clearly with the architecture.
+
+As an added benefits we take care of:
+
+-  ``x.to(device), y.to(device)``
+-  ``loss.backward()``
+-  ``optimizer.zero_grad()``
+-  ``optimizer.step()``
+
+As well as other features such as distributed training, numerics,
+gradient accumulation.
+
+.. code:: python
+
+   import torchvision
+   import torch.nn.functional as F
+
+   from composer.models import ComposerModel
+
+   class ResNet18(ComposerModel):
+
+       def __init__(self):
+           super().__init__()
+           self.model = torchvision.models.resnet18()
+
+       def forward(self, batch): # batch is the output of the dataloader
+           # specify how batches are passed through the model
+           inputs, _ = batch
+           return self.model(inputs)
+
+       def loss(self, outputs, batch):
+           # pass batches and `forward` outputs to the loss
+           _, targets = batch
+           return F.cross_entropy(outputs, targets)
+
+The composer model and can then be passed to our trainer.
+
+.. code:: python
+
+   from composer import Trainer
+
+   model = ResNet18()
+   optimizer = optim.Adam(model.parameters(), lr=0.0001)
+   train_dataloader # standard pytorch dataloader
+
+   trainer = Trainer(model=model, optimizers=optimizer,
+                     train_dataloader=train_dataloader, max_duration='10ep')
+   trainer.fit()
+
+Both the |forward| and |loss| methods are provided the ``batch`` directly
+from the dataloader. We leave the unpacking of that batch into inputs and targets
+to the user, since it can very depending on the task.
+
+We also provide several common classes for various tasks,
+specifically:
+
+-  :class:`.ComposerClassifier` - classification tasks with a cross entropy
+   loss and accuracy metric.
+-  :class:`.TIMM` - creates classification models from the popular `TIMM`_
+   library.
+-  :class:`.ComposerTransformer` - base class for 🤗 `Transformers`_ models.
+
+..note ::
+
+    Users from other frameworks such as pytorch lightning, may be used to
+    defining a ``training_step`` method, which groups the forward and loss
+    together. However, many of our algorithmic methods (such as
+    label smoothing or selective backprop) need to intercept and modify the
+    loss. For this reason, we split into two separate methods.
+
+By convention, we define our pytorch layers in the ``self.model``
```

Comment:
I don't think this paragraph is needed, since nothing depends on how the nn architecture is defined.